src/plasma/plasma_manager.cc

```diff
@@ -1031,7 +1065,11 @@ void object_table_subscribe_callback(ObjectID object_id,
       db_client_table_get_ip_addresses(manager_state->db, manager_ids);
   /* Run the callback for fetch requests if there is a fetch request. */
   auto it = manager_state->fetch_requests.find(object_id);
-  if (it != manager_state->fetch_requests.end()) {
+
+  if (it != manager_state->fetch_requests.end() &&
+      !is_receiving_or_received(manager_state, object_id)) {
```

Comment:
I would move the `is_receiving_or_received` check inside `request_transfer` when deciding whether to call `request_transfer_from`, so that we still record the manager locations received. Perhaps we should also rethink the naming of `request_transfer`...